src/components/pages/deals/Deals.js

```diff
@@ -20,14 +34,83 @@ const Deals = () => {
         fetch_deals()
     }, [])
 
-    const deals_to_display = isLoading === false && deals.length === 0 ? <div className="noDeals"><h1>No Deals available at the moment</h1></div> : <ProductsOnDeals items={deals}/>
+    // this function recieves the search text from the searchbar component and 
+    // sets it to the filter state and filters out some items depending on the text
+    const handleSearchedItems = (filter) => {
+
+        setSearchFilter(filter)
+
+        // this const is filtered items
+        const itemsSearched = deals.filter(deal => {
+            return deal.name.toLowerCase().includes(filter.toLowerCase())
+        })
+
+        setSearchedItems(itemsSearched)
+
+        // Each time a search it fined, this if determines wether to show products or
+        // show no search found by toggling the noItems state
+        if (searchFilter !== '' && itemsSearched.length === 0) {
+            setNoItems(true)
+        } else {
+            setNoItems(false)
+        }
+    }
+
+    const handleNextPage = () => {
+        setCurrentPageNumber(currentPageNumber+1)
+    }
+
+    const handlePreviousPage = () => {
+        setCurrentPageNumber(currentPageNumber - 1)
+    }
+
+    const handleFirstPage = () => {
+        setCurrentPageNumber(0)
+    }
+
+    const handleLastPage = () => {
+        setCurrentPageNumber(deals_chuncks.length - 1)
+    }
+
+    // groups the items to display into groups of 6
+    const number_of_items_to_display_per_page = 6
+    const deals_chuncks = _.chunk(deals, number_of_items_to_display_per_page)
```

Comment:
Use the API to query the data for the page instead of filtering the list of data you already have. Use the `from` and `size` query params.

If you just filter the data like this, you won't get all of the items in the store, because you are only filtering the 25 items that you loaded initially.